spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/AsyncRabbitTemplate.java

```diff
@@ -0,0 +1,589 @@
+			MessagePostProcessor messagePostProcessor) throws AmqpException {
+		return convertSendAndReceive(this.template.getExchange(), routingKey, message,
+				messagePostProcessor);
+	}
+
+	/**
+	 * Send a message to the supplied exchange and routing key. If the message
+	 * contains a correlationId property, it must be unique.
+	 * @param exchange the exchange.
+	 * @param routingKey the routing key.
+	 * @param message the message.
+	 * @return the {@link RabbitMessageFuture}.
+	 */
+	public RabbitMessageFuture sendAndReceive(String exchange, String routingKey, Message message) {
+		String correlationId = getOrSetCorrelationIdAndSetReplyTo(message);
+		RabbitMessageFuture future = new RabbitMessageFuture(correlationId);
+		CorrelationData correlationData = null;
+		if (this.enableConfirms) {
+			correlationData = new CorrelationData(correlationId);
+			future.setConfirm(new SettableListenableFuture<Boolean>());
+		}
+		this.pending.put(correlationId, future);
+		this.template.send(exchange, routingKey, message, correlationData);
+		return future;
+	}
+
+	/**
+	 * Convert the object to a message and send it to the provided exchange and
+	 * routing key after invoking the {@link MessagePostProcessor}.
+	 * If the post processor adds a correlationId property, it must be unique.
+	 * @param exchange the exchange
+	 * @param routingKey the routing key.
+	 * @param message the message.
+	 * @param messagePostProcessor the post processor.
+	 * @return the {@link RabbitConverterFuture}.
+	 */
+	public RabbitConverterFuture convertSendAndReceive(String exchange, String routingKey, Object message,
+			MessagePostProcessor messagePostProcessor) throws AmqpException {
+		CorrelationData correlationData = null;
+		if (this.enableConfirms) {
+			correlationData = new MutableCorrelationData(null);
+		}
+		CorrelationMessagePostProcessor correlationPostProcessor = new CorrelationMessagePostProcessor(
+				messagePostProcessor, correlationData);
+		this.template.convertAndSend(exchange, routingKey, message, correlationPostProcessor, correlationData);
+		return correlationPostProcessor.getFuture();
+	}
+
+	@Override
+	public synchronized void start() {
+		if (!this.running) {
+			this.container.start();
+		}
+		this.running = true;
+	}
+
+	@Override
+	public synchronized void stop() {
+		if (this.running) {
+			this.container.stop();
+		}
+		this.running = false;
+	}
+
+	@Override
+	public boolean isRunning() {
+		return this.running;
+	}
+
+	@Override
+	public int getPhase() {
+		return this.phase;
+	}
+
+	@Override
+	public boolean isAutoStartup() {
+		return this.autoStartup;
+	}
+
+	@Override
+	public void stop(Runnable callback) {
+		stop();
+		callback.run();
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public void onMessage(Message message) {
+		String correlationId = new String(message.getMessageProperties().getCorrelationId(), this.charset);
+		if (correlationId != null) {
```

Comment:
A message from the listener container will always have properties, but I agree with your comment.